engine/ebpf_compatibility_checker.rb

```diff
@@ -0,0 +1,56 @@
+require 'json'
+require 'open3'
+
+class EbpfCompatibilityChecker
+  attr_reader :system_information, :checked
+
+  def initialize(working_dir)
+    @working_dir = working_dir
+    @system_information = nil
+    @checked = false
+    check_compatibility
+  end
+
+  def mark_as_reported
+    @reported = true
+  end
+
+  def reported?
+    @reported || false
+  end
+
+  private
+
+  def check_compatibility
+    ebpf_script = File.join(@working_dir, 'ebpf.sh')
+
+    unless File.exist?(ebpf_script)
+      puts "eBPF compatibility check script not found at #{ebpf_script}"
+      @system_information = { error: "ebpf.sh script not found" }
+      return
+    end
+
+    begin
+      stdout, stderr, status = Open3.capture3("#{ebpf_script} --json")
+
+      if status.success?
+        @system_information = JSON.parse(stdout, symbolize_names: false)
+        @checked = true
+      else
+        puts "eBPF compatibility check failed with exit code #{status.exitstatus}"
+        puts "STDERR: #{stderr}" unless stderr.empty?
+        @system_information = {
+          error: "eBPF check failed",
+          exit_code: status.exitstatus,
+          stderr: stderr
```

Comment:
I'd capture stdout here too, just in case - doesn't look like we're redirecting it